bson/src/main/org/bson/codecs/configuration/CodecRegistry.java

```diff
@@ -26,9 +26,11 @@
  * {@code Object.equals}. It is not necessary to do so, and the simplest course of action is to rely on Object's implementation, but the
  * implementer may wish to implement a "value comparison" in place of the default "reference comparison."</p>
  *
+ * <p>As of the 4.0 release, this class extends the {@code CodecProvider} interface.  This capability was introduced to enable nesting
+ * registries inside another registry.</p>
  * @since 3.0
  */
-public interface CodecRegistry {
+public interface CodecRegistry extends CodecProvider {
```

Comment:
The boundaries are now really blurred between the two types. All CodecRegistries are now CodecProviders and all CodecProviders are not really CodecRegistries but they have the ability to be one because the `get` method provides the means to be one.

Reply:
I'm not sure what you mean.  Do you mean `CodecRegistry` or `CodecRegistries`?  Which `get` method are you referring to?